static/app/components/showHidePanel.tsx

```diff
@@ -0,0 +1,59 @@
+import {ReactNode, useState} from 'react';
+import styled from '@emotion/styled';
+
+import {Button} from 'sentry/components/button';
+import {IconChevron} from 'sentry/icons';
+import {t} from 'sentry/locale';
+import {space} from 'sentry/styles/space';
+
+interface Props {
+  content: () => ReactNode;
+  expandedByDefault: boolean;
+  headerHide: () => ReactNode;
+  headerShow: () => ReactNode;
```

Comment:
If you require dynamic text, you should leverage inversion of control and instead of defining a separate show/hide functions, pass the isShown value as the argument to a "header" function. _That said_, since your type is only limited to two values, you could also just pass a const string like "showLabel": string, "hideLabel": string